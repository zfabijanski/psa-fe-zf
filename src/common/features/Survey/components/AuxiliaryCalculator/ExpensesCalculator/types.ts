export interface IRow {
  header: string;
  suffix: string;
  index: number;
}
