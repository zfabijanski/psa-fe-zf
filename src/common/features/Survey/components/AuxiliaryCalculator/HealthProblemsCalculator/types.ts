export interface IHealthProblemData {
  image: string;
  title: string;
  amount: number;
  amountText: string;
  details: string[];
}

export interface IHealthProblem {
  data: IHealthProblemData[];
  onSave: (sum: number) => void;
  onCancel: () => void;
}
