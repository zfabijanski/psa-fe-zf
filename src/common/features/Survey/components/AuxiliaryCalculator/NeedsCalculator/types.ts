export interface IProduct {
  cost: number;
  desc: string;
  icon: JSX.Element;
}

export interface IRow {
  header?: string;
  products: IProduct[];
}
