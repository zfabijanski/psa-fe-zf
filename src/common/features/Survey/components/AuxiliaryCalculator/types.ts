import { IRow as IExpensesRow } from "./ExpensesCalculator/types";
import { IHealthProblemData } from "./HealthProblemsCalculator/types";
import { IRow as INeedsRow } from "./NeedsCalculator/types";

export type CalculatorData =
  | INeedsRow[]
  | IExpensesRow[]
  | IHealthProblemData[];

export type Calculator = "needs" | "expenses" | "healthProblems";

export interface ICalculatorData {
  data: CalculatorData;
  type: Calculator;
}
