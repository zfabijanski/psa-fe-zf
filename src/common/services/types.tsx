export interface IReport {
  quse_anse_map_code: string;
  value: string;
}
