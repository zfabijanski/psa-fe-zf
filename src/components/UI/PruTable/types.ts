import { ReactNode } from "react";

export type SortOrder = "descend" | "ascend";
export interface IColumn<T> {
  field: keyof T;
  title: string;
  width?: string;
  render?: (record: T) => ReactNode;
  sorter?: (a: T, b: T) => number;
}

export interface IColumnSorting<T> {
  field: keyof T;
  order: SortOrder;
}
