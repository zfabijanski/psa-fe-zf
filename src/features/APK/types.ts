export type IIncomeSource = "fullTimeOrContract" | "other" | "unemployed";
export interface ISetQuestionValue {
  setQuestionValue: (
    question: string,
    value?: string | number | boolean
  ) => void;
}

export interface IClientNeed {
  moveValueToCalculator?: boolean;
}

export interface INeedContainer {
  isExpandable: boolean;
}

export interface IPreliminary {
  incomeValue?: number;
  incomeSource?: IIncomeSource;
}

export interface ILifeInsurance extends IClientNeed {
  commitmentValue?: number;
  duration?: number;
  incomeForCollateral?: number;
  recommendedValue?: number;
  shouldRecalculateIncomeForCollateral: boolean;
}
export interface ISavings extends IClientNeed {
  valueToSave?: number;
  ownedSavings?: number;
  missingValue?: number;
}

export interface IChildFuture extends IClientNeed {
  expectedValue?: number;
  currentValue?: number;
  missingValue?: number;
}

export interface IRetirement extends IClientNeed {
  increasedBy?: number;
  duration?: number;
  pension?: number;
  gap?: number;
  recommendedValue?: number;
}

export interface IHealthProblems extends IClientNeed {
  incomeForCollateral?: number;
  hospitalStayInsuranceSumRecommendedValue?: number;
  hospitalStayForOneDayRecommendedValue?: number;
  seriousIllnessRecommendedValue?: number;
  seriousDisabilityRecommendedValue?: number;
  autoRecalculationEnabled?: boolean;
  shouldRecalculateSeriousIllness: boolean;
}
