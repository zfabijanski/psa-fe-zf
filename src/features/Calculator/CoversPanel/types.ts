import { ProductCategory } from "../../../models/common";
import { ICalculationVM } from "../types";

export interface IProductOption {
  productName: string;
  productCategories: ProductCategory[];
  initialValues: ICalculationVM;
}
