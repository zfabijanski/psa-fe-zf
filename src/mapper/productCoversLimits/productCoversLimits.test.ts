import { ProductGuid } from "features/Products/types";
import {
  mapProductsConfigToProductCoversLimits,
  calculateMultiplicity,
  toNumberOrUndefined,
  emptyCoverLimits,
} from "./productCoversLimits";
import { createProductConfig } from "testUtils";

describe("Function mapProductsConfigToProductCoversLimits", () => {
  it("should map products config to product covers limits", () => {
    expect(
      mapProductsConfigToProductCoversLimits([createProductConfig()])
    ).toMatchSnapshot();
  });

  it("should calculate multiplicity", () => {
    expect(calculateMultiplicity(ProductGuid.Start, "TPDISCH")).toBe(500);
    expect(calculateMultiplicity(ProductGuid.Start, "CICHILD")).toBe(500);
    expect(calculateMultiplicity(ProductGuid.Start, "HRC")).toBe(100);
    expect(calculateMultiplicity(ProductGuid.Start, "ABCD")).toBe(1000);
    expect(calculateMultiplicity(ProductGuid.Comfort, "HRA")).toBe(100);
    expect(calculateMultiplicity(ProductGuid.Comfort, "HRC")).toBe(100);
    expect(calculateMultiplicity(ProductGuid.Comfort, "ABCD")).toBe(1000);
    expect(calculateMultiplicity(ProductGuid.Savings, "HRA")).toBe(100);
    expect(calculateMultiplicity(ProductGuid.Savings, "ABCD")).toBe(1000);
    expect(calculateMultiplicity(ProductGuid.Retirement, "HRA")).toBe(100);
    expect(calculateMultiplicity(ProductGuid.Retirement, "ABCD")).toBe(1000);
    expect(calculateMultiplicity(ProductGuid.Oxygen, "TPDISCH")).toBe(1000);
  });

  it("should convert string to number or undefined", () => {
    expect(toNumberOrUndefined("123")).toBe(123);
    expect(toNumberOrUndefined("abc")).toBe(undefined);
  });
});

describe("emptyCoverLimits", () => {
  it("should return empty cover limits", () => {
    expect(emptyCoverLimits).toMatchInlineSnapshot(`
      Object {
        "addSumAssured": Object {},
        "duration": Object {},
        "expMatBen": Object {},
        "insuredAge": Object {},
        "payments": Object {},
        "policyHolderAge": Object {},
        "premium": Object {},
        "sumAssured": Object {},
        "sumOfPremiums": Object {},
      }
    `);
  });
});
