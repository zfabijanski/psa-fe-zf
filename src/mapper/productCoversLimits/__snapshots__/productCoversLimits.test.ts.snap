// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Function mapProductsConfigToProductCoversLimits should map products config to product covers limits 1`] = `
Object {
  "1": Object {},
}
`;
